packages/tasks/src/local-apps.ts

```diff
@@ -0,0 +1,119 @@
+import type { ModelData } from "./model-data";
+import type { PipelineType } from "./pipelines";
+
+/**
+ * Elements configurable by a local app.
+ */
+export type LocalApp = {
+	/**
+	 * Name that appears in buttons
+	 */
+	prettyLabel: string;
+	/**
+	 * Link to get more info about a local app (website etc)
+	 */
+	docsUrl: string;
+	/**
+	 * main category of app
+	 */
+	mainTask: PipelineType;
+	/**
+	 * Whether to display a pill "macOS-only"
+	 */
+	macOSOnly?: boolean;
+
+	comingSoon?: boolean;
+	/**
+	 * IMPORTANT: function to figure out whether to display the button on a model page's main "Use this model" dropdown.
+	 */
+	displayOnModelPage: (model: ModelData) => boolean;
+} & (
+	| {
+			/**
+			 * If the app supports deeplink, URL to open.
+			 */
+			deeplink: (model: ModelData) => URL;
+	  }
+	| {
+			/**
+			 * And if not (mostly llama.cpp), snippet to copy/paste in your terminal
+			 */
+			snippet: (model: ModelData) => string;
+	  }
+);
+
+function isGgufModel(model: ModelData) {
+	return model.tags.includes("gguf");
+}
+
+const snippetLlamacpp = (model: ModelData): string => {
+	return `llama-cpp \
+	--hf-repo ${model.id} \
+	-m file.gguf \
```

Comment:
```suggestion
	-m {FILE_NAME}.gguf \
```

Reply:
we don't have any file name here so i think it's better to write file.gguf

Reply:
Noting this here that this would be interesting to solve for, specially when it comes to multi-quants repo i.e. a repo with more than one quant.